expo/gbe/views/edit_volunteer_view.py

```diff
@@ -33,44 +39,81 @@ def EditVolunteerView(request, volunteer_id):
+        for interest in the_bid.volunteerinterest_set.all():
+            interest_form = VolunteerInterestForm(
+                request.POST,
+                instance=interest,
+                initial={'interest': interest.interest},
+                prefix=str(interest.pk))
+            formset += [interest_form]
+            if interest_form.is_valid():
+                if int(interest_form.cleaned_data.get('rank')) > 3:
+                    like_one_thing = True
+            else:
+                valid_interests = False
+
+        if form.is_valid() and valid_interests and like_one_thing:
             the_bid = form.save(commit=True)
             the_bid.available_windows.clear()
             the_bid.unavailable_windows.clear()
             for window in form.cleaned_data['available_windows']:
                 the_bid.available_windows.add(window)
             for window in form.cleaned_data['unavailable_windows']:
                 the_bid.unavailable_windows.add(window)
+            for interest_form in formset:
+                interest_form.save()
+
             user_message = UserMessage.objects.get_or_create(
                 view='EditVolunteerView',
                 code="SUBMIT_SUCCESS",
                 defaults={
                     'summary': "Volunteer Edit Success",
                     'description': default_volunteer_edit_msg})
             messages.success(request, user_message[0].description)
-            return HttpResponseRedirect(reverse('volunteer_review',
-                                                urlconf='gbe.urls'))
+            return HttpResponseRedirect("%s?conf_slug=%s" % (
+                reverse('volunteer_review', urlconf='gbe.urls'),
+                the_bid.conference.conference_slug))
+
         else:
+            formset += [form]
+            if not like_one_thing:
+                user_message = UserMessage.objects.get_or_create(
+                    view='EditVolunteerView',
+                    code="NO_INTERESTS_SUBMITTED",
+                    defaults={
+                        'summary': "Volunteer Has No Interests",
+                        'description': default_volunteer_no_interest_msg})
+                messages.error(request, user_message[0].description)
+
             return render(request,
                           'gbe/bid.tmpl',
-                          {'forms': [form],
+                          {'forms': formset,
                            'page_title': page_title,
                            'view_title': view_title,
                            'nodraft': 'Submit'})
     else:
-        if len(the_bid.interests.strip()) > 0:
-            interests_initial = eval(the_bid.interests)
-        else:
-            interests_initial = []
-        form = VolunteerBidForm(
+        # originally the volunteer create was supposed to save a title, there
+        # was a bug and most old bids don't have a title.
+        # This is the autocorrect
+        if not the_bid.title:
```

Comment:
I dig it.  I'll work on fixing that, and I *think* that's my last update on this before I request a re-review.